### PR TITLE
Update GCI mounter script to run in a rkt container

### PR DESCRIPTION
**Pull request by vishh**
Depends on #35652 

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35349)

### CODE REVIEW

**Comment by vishh** [comment]
@k8s-bot test this

File: cluster/gce/gci/mounter/mounter [hunk 1]
```diff
@@ -17,4 +17,20 @@
 set -e
 set -o pipefail
 
-sudo /bin/mount "$@"
\ No newline at end of file
+MOUNTER_DOCKER_IMAGE=gcr.io/google_containers/gci-mounter
+MOUNTER_DOCKER_VERSION=v1
+MOUNTER_USER=root
+RKT_BINARY=/home/kubernetes/bin/rkt
```
Comment (mtaufen) [inline cluster/gce/gci/mounter/mounter]: This won't work with the node e2e tests yet for a few reasons:
-  `/home/kubernetes/...` isn't a thing in node e2e
- node e2e GCI images don't have rkt installed anyway
- node e2e tests won't run the `cluster/gce/gci/configure.sh` script when they spin up a testing node, so https://github.com/kubernetes/kubernetes/pull/35321 won't have any effect on them.

Probably the easiest hack to fix this is to have the node test runner just create a `/home/kubernetes/` and drop `rkt` in there.

Comment (mtaufen) [inline cluster/gce/gci/mounter/mounter]: Normal e2es should be fine though

Comment (vishh) [inline cluster/gce/gci/mounter/mounter]: #35321 should have resolved the issue with node e2e.

**Comment by mtaufen** [comment]
For the GKE smoke, might be relevant:
`W1026 14:38:09.336] 2016/10/26 14:38:09 e2e.go:179: Something went wrong: encountered 3 errors: [error running get nodes: exit status 1 error running kubectl version: exit status 1 error running get status: exit status 1]`

^^not very many occurrences of "error" or "failed" in the build log for the GKE smoke.

For the GCE Node e2e, search for " Failed |" (or " Failed[0m |" if you're in the raw build log) and you see each reply from ginkgo is "0 Failed". So looks like the tests might be ok and the failure there is from something else.

**Comment by mtaufen** [comment]
LGTM

**Comment by vishh** [comment]
@k8s-bot node e2e test this

**Comment by jessfraz** [comment]
@vishh it looks like in trying to cherrypick this a lot of the logic doesnt currently exist on the 1.4 branch is that ok?

**Comment by yujuhong** [comment]
I saw repeated error messaged in kubelet.log in one of the PR builds and then kubelet got stuck and got killed:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/35572/pull-kubernetes-e2e-gce-gci/1817/

kubelet.log
```
Mounting command: /home/kubernetes/bin/mounter
Mounting arguments: tmpfs /media/root/var/lib/kubelet/pods/6c1435fd-9cde-11e6-8f2a-42010af00002/volumes/kubernetes.io~secret/default-token-x3w3h tmpfs []
Output: image: using image from local store for image name coreos.com/rkt/stage1-fly:1.17.0
image: using image from local store for url docker://gcr.io/google_containers/gci-mounter:v1
run: group "rkt" not found, will use default gid when rendering images
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/dev Fs:none Flags:1064960}
run: Processing {HostPath:/dev TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/dev Fs:none Flags:20480}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/proc Fs:none Flags:1064960}
run: Processing {HostPath:/proc TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/proc Fs:none Flags:20480}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/sys Fs:none Flags:1064960}
run: Processing {HostPath:/sys TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/sys Fs:none Flags:20480}
run: Processing {HostPath:tmpfs TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/tmp Fs:tmpfs Flags:0}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/ Fs:none Flags:1064960}
run: Processing {HostPath:/ TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/media/root Fs:none Flags:20480}
E1028 07:16:29.556787    1355 nestedpendingoperations.go:253] Operation for "\"kubernetes.io/secret/6c1435fd-9cde-11e6-8f2a-42010af00002-default-token-x3w3h\" (\"6c1435fd-9cde-11e6-8f2a-42010af00002\")" failed. No retries permitted until 2016-10-28 07:16:30.056752427 +0000 UTC (durationBeforeRetry 500ms). Error: MountVolume.SetUp failed for volume "kubernetes.io/secret/6c1435fd-9cde-11e6-8f2a-42010af00002-default-token-x3w3h" (spec.Name: "default-token-x3w3h") pod "6c1435fd-9cde-11e6-8f2a-42010af00002" (UID: "6c1435fd-9cde-11e6-8f2a-42010af00002") with: mount failed: exit status 1
Mounting command: /home/kubernetes/bin/mounter
Mounting arguments: tmpfs /media/root/var/lib/kubelet/pods/6c1435fd-9cde-11e6-8f2a-42010af00002/volumes/kubernetes.io~secret/default-token-x3w3h tmpfs []
Output: image: using image from local store for image name coreos.com/rkt/stage1-fly:1.17.0
image: using image from local store for url docker://gcr.io/google_containers/gci-mounter:v1
run: group "rkt" not found, will use default gid when rendering images
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/dev Fs:none Flags:1064960}
run: Processing {HostPath:/dev TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/dev Fs:none Flags:20480}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/proc Fs:none Flags:1064960}
run: Processing {HostPath:/proc TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/proc Fs:none Flags:20480}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/sys Fs:none Flags:1064960}
run: Processing {HostPath:/sys TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/sys Fs:none Flags:20480}
run: Processing {HostPath:tmpfs TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/tmp Fs:tmpfs Flags:0}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/ Fs:none Flags:1064960}
run: Processing {HostPath:/ TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/media/root Fs:none Flags:20480}
I1028 07:16:29.556810    1355 server.go:636] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"l7-default-backend-v1.0-kow7y", UID:"6c1435fd-9cde-11e6-8f2a-42010af00002", APIVersion:"v1", ResourceVersion:"195", FieldPath:""}): type: 'Warning' reason: 'FailedMount' MountVolume.SetUp failed for volume "kubernetes.io/secret/6c1435fd-9cde-11e6-8f2a-42010af00002-default-token-x3w3h" (spec.Name: "default-token-x3w3h") pod "6c1435fd-9cde-11e6-8f2a-42010af00002" (UID: "6c1435fd-9cde-11e6-8f2a-42010af00002") with: mount failed: exit status 1
Mounting command: /home/kubernetes/bin/mounter
Mounting arguments: tmpfs /media/root/var/lib/kubelet/pods/6c1435fd-9cde-11e6-8f2a-42010af00002/volumes/kubernetes.io~secret/default-token-x3w3h tmpfs []
Output: image: using image from local store for image name coreos.com/rkt/stage1-fly:1.17.0
image: using image from local store for url docker://gcr.io/google_containers/gci-mounter:v1
run: group "rkt" not found, will use default gid when rendering images
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/dev Fs:none Flags:1064960}
run: Processing {HostPath:/dev TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/dev Fs:none Flags:20480}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/proc Fs:none Flags:1064960}
run: Processing {HostPath:/proc TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/proc Fs:none Flags:20480}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/sys Fs:none Flags:1064960}
run: Processing {HostPath:/sys TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/sys Fs:none Flags:20480}
run: Processing {HostPath:tmpfs TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/tmp Fs:tmpfs Flags:0}
run: Processing {HostPath: TargetPrefixPath: RelTargetPath:/ Fs:none Flags:1064960}
run: Processing {HostPath:/ TargetPrefixPath:stage1/rootfs/opt/stage2/gci-mounter/rootfs RelTargetPath:/media/root Fs:none Flags:20480}
```

**Comment by yujuhong** [comment]
@mtaufen, I found failures with similar messages:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/31647/pull-kubernetes-e2e-gce-gci/1814/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/31647/pull-kubernetes-e2e-gce-gci/1816/

**Pull request merged**